### PR TITLE
ROC-5098: Use elastic search for get cases

### PR DESCRIPTION
**Pull request by dharmendrak**
Jira: 
https://tools.hmcts.net/jira/browse/ROC-5098

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,7 +11,7 @@ plugins {
 }
 
 group 'uk.gov.hmcts.reform'
-version '4.4.1'
+version '4.4.2-SANPSHOT'
```
Comment (markry-kainos) [inline build.gradle]: SNAP :)

File: src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    private final ObjectMapper mapper;
+
+    public SearchCriteria(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public String matchAllQuery() {
+        JsonNode query = mapper.createObjectNode()
+            .put("size", 10)
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java]: Can we make this optional that defaults to 10 if not provided?

File: src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            .set("query", mapper.createObjectNode()
+                .set("match_all", mapper.createObjectNode()));
+
+        return query.toString();
+    }
+
+    public String searchByQuery(String key, String value) {
+        JsonNode query = mapper.createObjectNode()
+            .put("size", 10)
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java]: what happens if there is more than 10 results?

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java]: Is this `size` param to override default value of `NN` that CCD uses ? If that's the case then:

1. if `size` is not passed then use the default of CCD ( which means don't need to pass size at all in our query)
2. if we want to override it, can pass `size`

File: build.gradle (hunk 1 above)
Comment (sabahirfan) [reply build.gradle]: I think you need to bump up the Y part of the version (X.Y.Z)

Comment (markry-kainos) [reply build.gradle]: yeah - this is a good point - maybe should bump major version X (semver = incompatible changes...is it? if not then fine)

File: src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    private final ObjectMapper mapper;
+
+    public SearchCriteria(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public String matchAllQuery(Integer size) {
+        JsonNode query = mapper.createObjectNode()
+            .put("size", Optional.ofNullable(size).orElse(MINIMUM_SIZE_PER_PAGES))
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java]: Suggestion:
`The above forces the caller to pass either null or some value. which is not very user friendly.`

we can have overloaded methods like:
```
public String matchAllQuery(Integer size) {
...
.put("size",  size)
...
```
and other one:

```
public String matchAllQuery() {
return matchAllQuery(MINIMUM_SIZE_PER_PAGES)
}
```

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/ccd/client/model/SearchCriteria.java]: Thank you very much @dharmendrak  👍

File: build.gradle (hunk 1 above)
Comment (sabahirfan) [reply build.gradle]: This is new backward compatible functionality , so don't need to bump X.

**Pull request merged**